### PR TITLE
Fixes 51

### LINKED ISSUE #51: Delete object is broken
Deleting an object actually deletes another object:

https://user-images.githubusercontent.com/4711805/205687829-e31c75ba-e48a-42d0-a289-dc138189ec75.mp4

### PR DESCRIPTION
**Pull request by hbcarlos**
Fixes #51 

Fixes removing objects and adding objects with duplicated names.

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -256,6 +256,14 @@ export class JupyterCadDoc
     return this._metadataChanged;
   }
 
+  objectExists(name: string): boolean {
+    const obj = this._getObjectAsYMapByName(name);
+    if (obj) {
+      return true;
+    }
+    return false;
```
Comment (martinRenou) [inline src/model.ts:264]:
```suggestion
    return Boolean(this._getObjectAsYMapByName(name));
```

File: src/toolbar/operatortoolbar.tsx [hunk 2]
```diff
@@ -111,7 +126,14 @@ export class OperatorToolbarReact extends React.Component<IProps> {
           const model = this.props.toolbarModel.sharedModel;
           if (model) {
             model.transact(() => {
-              model.addObject(objectModel);
+              if (!model.objectExists(objectModel.name)) {
+                model.addObject(objectModel);
```
Comment (martinRenou) [inline src/toolbar/operatortoolbar.tsx:130]: Should we make the `objectExists` check under `addObject` instead? So we don't need to add that check code everywhere we use `addObject`?

Comment (hbcarlos) [reply src/toolbar/operatortoolbar.tsx:130]: `addObject` also checks whether it exists or not. The reason I added `objectExists` is to be able to check before trying to add the element and show an error mesage if it exists.

**Comment by trungleduc** [comment]
Could you rebase it to pick up the CI fix?

**Review by trungleduc — APPROVED** [review]

**Comment by trungleduc** [comment]
Thanks @hbcarlos

**Pull request merged**